ha/cli/exec/clusterExecutor.py

```diff
@@ -129,50 +136,156 @@ def execute(self):
+                            type=self._is_file_extension_valid)
+        parser.add_argument('--username', help='cluster username', required=True)
+        parser.add_argument('--password', help='cluster password', required=True)
+        parser.add_argument('--json', help='Required output fomat', action='store_true')
+        self._args = parser.parse_args()
+        return True
+
+    def _is_file_extension_valid(self, filename) -> str:
+        '''
+           Checks if file extension which is passed is correct or not
+           Returns: str
+           Exception: ArgumentTypeError
+        '''
+        base, ext = os.path.splitext(filename)
+        if ext.lower() not in ('.json'):
+            raise argparse.ArgumentTypeError('File must have a json extension')
+        return filename
+
+    def validate(self) -> bool:
+        '''
+           Validates permission and command line arguments.
+           Exception: HAInvalidPermission
+           Return: bool
+        '''
+        # Every CLI command will be an internal command now. So,
+        # we do not need this change. If required, can be revisited later.
+        #if not self.is_ha_user():
+        #    raise HAInvalidPermission('Not enough permissions to invoke this command')
+        if self.parse_cluster_args():
+            return True
+        return False
+
+    def _is_valid_node_id(self, node_id) -> bool:
+        '''
+           Checks if node id gets resolved to some IP address or not
+           Returns: bool
+           Exception: socket.gaierror, socket.herror
+        '''
+        ip_validator_regex = "^((25[0-5]|2[0-4][0-9]|1[0-9][0-9]|[1-9]?[0-9])\.){3}(25[0-5]|2[0-4][0-9]|1[0-9][0-9]|[1-9]?[0-9])$"
+
+        # node_id can be passed as IP address or FQDN or
+        # some random number or just sequence of chars
+
+        # first seperate the string from dots.
+        splitted_node_id = node_id.replace('.', '')
+
+        # If string only contains numbers, means it can be just
+        # random number or can be an IP address. So, IP address
+        # validation can be done. else for random number, exception will be
+        # raised
+        if re.search('^[0-9]*$', splitted_node_id):
+            print(f'Validating the IP address: {node_id}')
+            if re.search(ip_validator_regex, node_id):
+                return True
+            raise HAClusterCLIError(f'{node_id} is not a valid node_id')
+        # else it can be combination of chars and numbers means hostname or just a
+        # random meaningless string
+        else:
+            print(f'Validating the hostname: {node_id}')
+            try:
+                resolved_ip = socket.gethostbyname(node_id)
+            except socket.gaierror as se:
+                raise HAClusterCLIError(f'{node_id} not a valid node_id: {se}')
+            except socket.herror as he:
+                raise HAClusterCLIError(f'{node_id} not a valid node_id: {he}')
+            except Exception as err:
```

Comment:
I think we can keep just this last exception handler since we are doing the same thing all three?

Reply:
Done.